src/skin.cpp

```diff
@@ -111,9 +112,32 @@ void generateSkinAreas(int layer_nr, SliceMeshStorage& mesh, const int innermost
                 upskin = upskin.difference(not_air); // skin overlaps with the walls
             }
         }
+
+        int expand_skins_expand_distance = mesh.getSettingInMicrons("expand_skins_expand_distance");
         
+        if (expand_skins_expand_distance > 0)
+        {
+            int expand_skins_shrink_distance = mesh.getSettingInMicrons("expand_skins_shrink_distance");
+
+            // skin areas are to be enlarged by expand_distance but before they are expanded
+            // the skin areas are shrunk by shrink_distance so that very narrow regions of skin
+            // (often caused by the model's surface having a steep incline) are removed first
+
+            expand_skins_expand_distance += expand_skins_shrink_distance; // increase the expansion distance to compensate for the shrinkage
+
+            if (mesh.getSettingBoolean("expand_upper_skins"))
+            {
+                upskin = upskin.offset(-expand_skins_shrink_distance).offset(expand_skins_expand_distance).intersection(virgin_upskin);
```

Comment:
Hi, adding the union stops small skin areas from being removed but that is exactly what some people are asking for (see https://ultimaker.com/en/community/39214-feature-add-angle-for-top-bottom-internal-layer-generation).

Perhaps we should add an option (something like Preserve Original Skin) to control whether we do the union or not?